compiler/vm/vm.nim

```diff
@@ -2546,18 +3127,22 @@ proc evalMacroCall*(module: PSym; idgen: IdGenerator; g: ModuleGraph; templInstC
   #InternalAssert tos.slots.len >= L
 
   # return value:
-  tos.slots[0] = TFullReg(kind: rkNode, node: newNodeI(nkEmpty, n.info))
+  doAssert false #tos.slots[0] = TFullReg(kind: rkNode, node: newNodeI(nkEmpty, n.info))
 
-  # setup parameters:
+  # put macro call arguments onto the register stack 
   for i in 1..<sym.typ.len:
     tos.slots[i] = setupMacroParam(n[i], sym.typ[i])
 
+  # TODO: is this doc comment correct?
+  # put macro generic parameters onto the register stack 
   let gp = sym.ast[genericParamsPos]
   for i in 0..<gp.len:
     let idx = sym.typ.len + i
     if idx < n.len:
       tos.slots[idx] = setupMacroParam(n[idx], gp[i].sym.typ)
     else:
+      # TODO: are the surrounding check and this block meant to be in the for-loop?
```

Comment:
Pretty sure they are, of everything works we never go into the else, if we do we cleanup (eval failed, hence dec) and then report an error

Reply:
But if `gp.len > 1` and `sym.typ.len == n.len` then we'd enter the else branch is taken multiple times, always reporting the same error and also decrementing `evalMacroCounter` each time.

From what I can tell, `evalMacroCounter` is incremented only once at the start of evaluating a macro and then decremented at the end. If the else branch is taken even once, `evalMacroCall` would return with an `evalMacroCounter` less than the one it was called with which doesn't make sense, so at least the `dec(g.config.evalMacroCounter)` is wrong